arcgis-ios-sdk-samples/Maps/Show location history/LocationHistoryViewController.swift

```diff
@@ -0,0 +1,191 @@
+//
+// Copyright © 2019 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+// MARK: - Constants
+
+private enum Constants {
+    static let locationBorderColor = UIColor.backgroundGray
+    static let locationInnerColor = UIColor.primaryTextColor
+    static let trackColor = UIColor.primaryBlue.withAlphaComponent(0.5)
+
+    static let locationDiameter: CGFloat = 14
+    static let locationBorderWidth: CGFloat = 2
+    static let trackWidth: CGFloat = 10
+
+    static let initialZoomScale: Double = 10000
+}
+
+// MARK: - LocationTrackingStatus
+
+private enum LocationTrackingStatus {
+    case paused
+    case inProgress
+}
+
+// MARK: - LocationHistoryViewController
+
+class LocationHistoryViewController: UIViewController {
+    @IBOutlet private weak var mapView: AGSMapView!
+    @IBOutlet private weak var trackingBarButtonItem: UIBarButtonItem!
+    
+    private var currentStatus: LocationTrackingStatus = .paused {
+        didSet {
+            handleLocationStatusChange()
+        }
+    }
+    
+    private let map = AGSMap(basemap: .lightGrayCanvasVector())
+
+    private let locationsOverlay: AGSGraphicsOverlay = {
+        let overlay = AGSGraphicsOverlay()
+
+        let locationSymbol = AGSSimpleMarkerSymbol(style: .circle, color: Constants.locationInnerColor, size: Constants.locationDiameter)
+        locationSymbol.outline = AGSSimpleLineSymbol(style: .solid, color: Constants.locationBorderColor, width: Constants.locationBorderWidth)
+
+        overlay.renderer = AGSSimpleRenderer(symbol: locationSymbol)
+
+        return overlay
+    }()
+    
+    private let trackOverlay: AGSGraphicsOverlay = {
+        let overlay = AGSGraphicsOverlay()
+
+        let trackSymbol = AGSSimpleLineSymbol(style: .solid, color: Constants.trackColor, width: Constants.trackWidth)
+        overlay.renderer = AGSSimpleRenderer(symbol: trackSymbol)
+
+        return overlay
+    }()
+
+    private var trackBuilder: AGSPolylineBuilder?
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupView()
+    }
+    
+    // MARK: IBActions
+    
+    @IBAction private func trackingTapped(_ sender: UIBarButtonItem) {
+        toggleTrackingStatus()
```

Comment:
I like the idea of modeling the tracking state with an enum, but I'm wondering if it unnecessarily complicates the sample. Since there are only two states, the tracking state could be modeled with a Boolean. That would allow you to use the built-in `toggle()` function here:

```swift
isTracking.toggle()
```

While it is possible that the sample could be updated in the future to introduce additional states, that is unlikely. Sample designs are fairly static.

Reply:
I incorporated this feedback via [`967da7c8`](https://github.com/Esri/arcgis-runtime-samples-ios/pull/751/commits/967da7c8953d3e213ef0dde931e15ed241cc7a5b).